src/Java.Interop.Tools.JavaTypeSystem/JavaModels/JavaMethodModel.cs

```diff
@@ -0,0 +1,188 @@
+using System;
+using System.Collections.Generic;
+using System.Linq;
+
+namespace Java.Interop.Tools.JavaTypeSystem.Models
+{
+	public class JavaMethodModel : JavaMemberModel
+	{
+		public string Return { get; }
+		public string ReturnGeneric { get; }
+		public bool IsAbstract { get; }
+		public bool IsBridge { get; }
+		public string ReturnJni { get; }
+		public bool IsSynthetic { get; }
+		public bool IsSynchronized { get; }
+		public bool IsNative { get; }
+		public bool ReturnNotNull { get; }
+
+		public JavaTypeParameters TypeParameters { get; }
+		public JavaTypeReference? ReturnTypeModel { get; private set; }
+		public JavaMethodModel? BaseMethod { get; set; }
+		public List<JavaParameterModel> Parameters { get; } = new List<JavaParameterModel> ();
+		public List<JavaExceptionModel> Exceptions { get; } = new List<JavaExceptionModel> ();
+
+		public JavaMethodModel (string javaName, string javaVisibility, bool javaAbstract, bool javaFinal, bool javaStatic, string javaReturn, JavaTypeModel javaParentType, string deprecated, string jniSignature, bool isSynthetic, bool isBridge, string returnJni, bool isNative, bool isSynchronized, bool returnNotNull)
+			: base (javaName, javaStatic, javaFinal, javaVisibility, javaParentType, deprecated, jniSignature)
+		{
+			IsAbstract = javaAbstract;
+			Return = javaReturn;
+			ReturnGeneric = javaReturn;
+			IsBridge = isBridge;
+			IsSynthetic = isSynthetic;
+			ReturnJni = returnJni;
+			IsNative = isNative;
+			IsSynchronized = isSynchronized;
+			ReturnNotNull = returnNotNull;
+
+			TypeParameters = new JavaTypeParameters (this);
+		}
+
+		public override void Resolve (JavaTypeCollection types, List<JavaUnresolvableModel> unresolvables)
+		{
+			if (Name.Contains ('$')) {
```

Comment:
Similar to `JavaFieldModel`, `$` is a valid Java character.  Why invalidate the method if it contains `$`?

Reply:
Compatibility with existing code: https://github.com/xamarin/java.interop/commit/835aaf773f9f7e21df51f4f64c99945f04dd4e0f.